external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py

```diff
@@ -193,40 +197,36 @@ def export(self, export_type: ExportType, output_model: ModelEntity):
-            output_model.set_data(
-                "label_schema.json",
-                label_schema_to_bytes(self._task_environment.label_schema),
-            )
-        logger.info("Exporting completed")
-
-    def _init_recipe_hparam(self) -> dict:
-        warmup_iters = int(self._hyperparams.learning_parameters.learning_rate_warmup_iters)
-        lr_config = (
-            ConfigDict(warmup_iters=warmup_iters)
-            if warmup_iters > 0
-            else ConfigDict(warmup_iters=warmup_iters, warmup=None)
-        )
-
-        if self._hyperparams.learning_parameters.enable_early_stopping:
-            early_stop = ConfigDict(
-                start=int(self._hyperparams.learning_parameters.early_stop_start),
-                patience=int(self._hyperparams.learning_parameters.early_stop_patience),
-                iteration_patience=int(self._hyperparams.learning_parameters.early_stop_iteration_patience),
+            output_model.set_data("label_schema.json", label_schema_to_bytes(self._task_environment.label_schema))
+            output_model.set_data("config.json", config_to_bytes(self._hyperparams))
+        logger.info('Exporting completed')
+
+    def _overwrite_parameters(self):
+        """ Overwrite config parameters with TaskEnvironment hyper-parameters and config tiling parameters. """
+        super()._overwrite_parameters()
+        if bool(self._hyperparams.tiling_parameters.enable_tiling):
+            logger.info("Tiling Enabled")
+            tile_params = ConfigDict(
+                data=ConfigDict(
+                    train=ConfigDict(
+                        tile_size=int(self._hyperparams.tiling_parameters.tile_size),
+                        overlap_ratio=float(self._hyperparams.tiling_parameters.tile_overlap),
+                        max_per_img=int(self._hyperparams.tiling_parameters.tile_max_number)
+                    ),
+                    val=ConfigDict(
+                        tile_size=int(self._hyperparams.tiling_parameters.tile_size),
+                        overlap_ratio=float(self._hyperparams.tiling_parameters.tile_overlap),
+                        max_per_img=int(self._hyperparams.tiling_parameters.tile_max_number)
+                    ),
+                    test=ConfigDict(
+                        tile_size=int(self._hyperparams.tiling_parameters.tile_size),
+                        overlap_ratio=float(self._hyperparams.tiling_parameters.tile_overlap),
+                        max_per_img=int(self._hyperparams.tiling_parameters.tile_max_number)
+                    )
+                )
             )
-        else:
-            early_stop = False
-
-        return ConfigDict(
-            optimizer=ConfigDict(lr=self._hyperparams.learning_parameters.learning_rate),
-            lr_config=lr_config,
-            early_stop=early_stop,
-            use_adaptive_interval=self._hyperparams.learning_parameters.use_adaptive_interval,
```

Comment:
This is a setting (use_adaptive_interval) that exists only for the current detection, but will it be maintained?

Reply:
@harimkang Thanks Harim, I missed it. It will be maintained: 

```
self._recipe_cfg.merge_from_dict(
    dict(use_adaptive_interval=self._hyperparams.learning_parameters.use_adaptive_interval))
```